org.eclipse.xtext.xbase.tests/src/org/eclipse/xtext/xbase/tests/AbstractXbaseTestCase.java

```diff
@@ -41,8 +43,13 @@ public static boolean isJava11OrLater() {
 	private static boolean determineJava11OrLater() {
 		String javaVersion = System.getProperty("java.version");
 		try {
-			int version = Integer.parseInt(javaVersion);
-			return version >= 11;
+			Pattern p = Pattern.compile("(\\d+)(.)*");
```

Comment:
I wonder if we should move this to xtext.util.JavaVersion?

Reply:
methodname?

Reply:
JavaVersion.getRuntimeVersion(): JavaVersion?

Reply:
well then i have to detect 1.4-1.8 too

Reply:
Hmm true. Not worth it as of now.

Reply:
will file a second issue, maybe you can give it a shot